src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs

```diff
@@ -120,5 +120,7 @@ private void RaiseOnException(Exception ex)
                 // Ignore exceptions from the exception handlers themselves.
             }
         }
+
+        internal virtual bool CanThrowExceptions => true;
```

Comment:
Sure, it is just a small perf optimization to avoid forcing Localizable string message for sub-types which are guaranteed to never throw exceptions.